DetectorDescription/DDCMS/plugins/DDDefinitions2Objects.cc

```diff
@@ -680,33 +678,71 @@ template <> void Converter<DDLPosPart>::operator()( xml_h element ) const {
 }
 
 /// Converter for <PartSelector/> tags
-template <> void Converter<PartSelector>::operator()( xml_h element ) const {
-  cms::DDNamespace ns( _param<cms::DDParsingContext>()); //, element, true );
-  xml_dim_t e( element );
+template <> void Converter<PartSelector>::operator()(xml_h element) const {
+  cms::DDNamespace ns(_param<cms::DDParsingContext>());
+  cms::DDParsingContext* const context = ns.context();
+  DDSpecParRegistry& registry = *context->description->extension<DDSpecParRegistry>();
+  xml_dim_t e(element);
   xml_dim_t specPar = e.parent();
-  string specParName = specPar.attr<string>( _U( name ));
-  string path = e.attr<string>( DD_CMU( path ));
-
+  string specParName = specPar.attr<string>(_U(name));
+  string path = e.attr<string>(DD_CMU(path)); 
   printout(ns.context()->debug_specpars ? ALWAYS : DEBUG, "DD4CMS",
            "+++ PartSelector for %s path: %s", specParName.c_str(), path.c_str());
+  registry.specpars[specParName].paths.emplace_back(path);
 }
 
 /// Converter for <Parameter/> tags
-template <> void Converter<Parameter>::operator()( xml_h element ) const {
-  cms::DDNamespace ns( _param<cms::DDParsingContext>()); //, element, true );
-  xml_dim_t e( element );
+template <> void Converter<Parameter>::operator()(xml_h element) const {
+  cms::DDNamespace ns(_param<cms::DDParsingContext>());
+  cms::DDParsingContext* const context = ns.context();
+  DDSpecParRegistry& registry = *context->description->extension<DDSpecParRegistry>();
+  xml_dim_t e(element);
+  xml_dim_t specPar = e.parent();
+  xml_dim_t specParSect = specPar.parent();
+  string specParName = specPar.attr<string>(_U(name));
   string name = e.nameStr();
-  string value = e.attr<string>( DD_CMU( value ));
-  bool eval = e.hasAttr( _U( eval )) ? e.attr<bool>( _U( eval )) : false;
+  string value = e.attr<string>(DD_CMU(value));
+  bool eval = e.hasAttr(_U(eval)) ? e.attr<bool>(_U(eval)) : (specParSect.hasAttr(_U(eval)) ? specParSect.attr<bool>(_U(eval)) : false);
```

Comment:
This line qualifies for a comment. You are setting variable based upon its own value???

Reply:
the variable is set based on an "eval" attribute of an element if it's present, or the parent's element attribute.

Reply:
`_U(eval)` actually becomes  `::dd4hep::xml::Unicode_eval`, which is a constant. Giving `bool eval` the same name as the tag is confusing. I suggest calling it `numberTag` since it seems to indicate a number when it is true.

Reply:
it is actually a flag which indicates if a string needs to be evaluated. The naming is done on purpose and is consistent throughout.